File: StudInfoWeb/wwwroot/js/student.js
var dataTable;

$(document).ready(function () {
    loadStudentTable();
})



function loadStudentTable() {
    dataTable = $("#tblStudent").DataTable({
        ajax: "/Student/GetAllStudents",
        columns: [
            { data: "studNos" },
            { data: "firstName" },
            { data: "lastName" },
            {
                data: "sex",
                render: function (data) {
                    return data == 0 ? "Male" : "Female"
                }
            },
            { data: "deptCode" },
            {
                data: "degreeProgramName",
                render: function (data) {
                    switch (data) {
                        case 1: return "Accounting"; break;
                        case 2: return "Business Administration"; break;
                        case 3: return "Biological Science"; break;
                        case 4: return "Computer Science"; break;
                        case 5: return "Electrical Electronic Eng"; break;
                        case 6: return "Mathematics"; break;
                        case 7: return "Medicine & Surgery"; break;
                    }
                }
            },
            {
                data: "id",
                render: function (data) {
                    return `
                             
                              <div class="btn-group" role="group">

                                <a href="/Student/Upsert?id=${data}"
                                class="btn btn-primary mx-2"> <i class="bi bi-pencil-square"></i>&nbsp;Edit</a>

                                <a href='/Student/Delete?id=${data}'
                                class="btn btn-danger mx-2"> <i class="bi bi-trash-fill"></i>&nbsp;Delete</a>

					        </div>
                        `
                }
            }
        ]
    })
}

File: StudInfoModel/Student.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;



namespace StudInfoModel
{
       
    public class Student
    {
        public int Id { get; set; }


        [Key]
        [Required]
        [StringLength(50)]
        [Display(Name = "Student Nos")]
        public string StudNos { get; set; }



        [Required]
        [Display(Name = "First Name")]
        [StringLength(80)]
        public string FirstName { get; set; }


        [Required]
        [Display(Name = "Last Name")]
        [StringLength(80)]
        public string LastName { get; set; }


        [Required]
        [StringLength(90)]
        [DataType(DataType.EmailAddress)]
        public string Email { get; set; }


        [Required]
        [StringLength(50)]
        [Display(Name = "Phone")]
        public string PhoneNos { get; set; }


        [Required]
        [Display(Name = "Date of Birth")]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:dd-MMM-yyyy}")]
        public DateTime DOB { get; set; }


        [Required]
        [Display(Name = "Admission Date")]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:dd-MMM-yyyy}")]
        public DateTime AdmissionDate { get; set; }



        [Required]
        public Gender Sex  { get; set; }


        [Display(Name = "Picture")]
        public byte[]? Pix { get; set; }


        [Required]
        [Display(Name = "Dept")]
        public string DeptCode { get; set; }


        [ForeignKey("DeptCode")]
        [ValidateNever]
        public Dept? Dept { get; set; }



        [Required]
        [Display(Name = "Degree Program Name")]
        public ProgramName DegreeProgramName { get; set; }


        [Required]
        [Display(Name = "Degree Title")]
        public DegreeTitle DegreeTitleToBeAwarded { get; set; }


        public int? Level { get; set; }


        public int? Semester { get; set; }


        [StringLength(20)]
        public string? LatestEnrollGrpCode { get; set; }


      
        public float? GPA { get; set; }


        
        public float? CGPA { get; set; }


        [StringLength(500)]
        public string? Remarks { get; set; }


        [StringLength(3000)]
        public string? PhotoBackupPath { get; set; }


    }



#region Enums


    public enum Gender
    {
        Male = 1,
        Female = 2
    }

    
    public enum ProgramName
    {
        Accounting = 1,

        [Display(Name = "Business Administration")]
        BusAdmin = 2,

        [Display(Name = "Biological Science")]
        BioScience = 3,

        [Display(Name = "Computer Science")]
        ComputerScience = 4,

        [Display(Name = "Electrical Electronics Engineering")]
        ElectElect = 5,


        Mathematics = 6,


        [Display(Name = "Medicine and Surgery")]
        MedSurgery = 7
    }



    public enum DegreeTitle {
        [Display(Name = "Bachelor of Art")]
        BA = 1,

        [Display(Name = "Bachelor of Engineering")]
        BEng = 2,

        [Display(Name = "Bachelor of Science")]
        BSc = 3,

        [Display(Name = "Bachelor of Medicine and Bachelor of Surgery")]
        MBBS = 4
    }


    

    #endregion Enums









}
